### ISSUE TITLE
z3str3 assertion violation at ../src/util/rational.h Line: 113

### ISSUE THREAD

**Issue by muchang** [Z3Prover/z3#2951]
Hi,
For this formula,
```
(declare-fun x () String)
(declare-fun y () String)
(assert (or (= (str.replace (str.replace x y x) x y) x)))
(check-sat)
```
`z3 smt.string_solver=z3str3 ` throws out an assertion violation:
```
ASSERTION VIOLATION
File: ../src/util/rational.h
Line: 113
is_unsigned()
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
```

OS: Ubuntu 18.04
Revision 140926e

**Comment by mtrberzi** [comment]
Taking a look, it seems like a string variable is incorrectly getting assigned the length -1 at some point in the search. We may be missing an axiom somewhere.

**Comment by mtrberzi** [comment]
Fixed in #3032
